src/scala/org/pantsbuild/zinc/compiler/Main.scala

```diff
@@ -187,18 +197,87 @@ object Main {
     Settings.SettingsParser
       .parse(preprocessArgs(context.getArgs), Settings()) match {
       case Some(settings) =>
+        val workingDirectory = new File(context.getWorkingDirectory)
         mainImpl(
-          settings.withAbsolutePaths(new File(context.getWorkingDirectory)),
+          settings.withAbsolutePaths(workingDirectory),
           startTime,
-          n => context.exit(n))
+          n => context.exit(n),
+          workingDirectory
+        )
       case None => {
         println("See zinc-compiler --help for information about options")
         context.exit(1)
       }
     }
   }
 
-  def mainImpl(settings: Settings, startTime: Long, exit: Int => Unit): Unit = {
+  def toLsp(severity: Severity): DiagnosticSeverity = {
```

Comment:
Having these live here feels weird. Could we maybe move them to one of the `util` packages?

Reply:
I see why you feel this way; but really the part of this which is general and re-usable is the LSP types which I'm getting from lsp4j.
This conversion from xsbti types to LSP is just needed here for now, and I don't think has a lot of meaning outside of this context. I'd rather keep it local for now, with the possibility of extracting it to somewhere more general later if I'm proven wrong and we do need these conversions from somewhere else later.